### PR TITLE
Use read by index and check for NaNs

### PR DESCRIPTION
**Pull request by ahiguera-mx**
_Before you submit this PR: make sure to put all operations-related information in a wiki-note, a PR should be about code and is publicly accessible_
Use read by index and check for NaNs
## What does the code in this PR do / what does it improve?
Use read by index and check for NaNs see https://github.com/AxFoundation/strax/pull/529
## Can you give a minimal working example (or illustrate with a figure)?

_Please include the following if applicable:_
  - [ x] _Update the docstring(s)_
  - [x ] _Update the documentation_
  - [x ] _Tests to check the (new) code is working as desired._
  - [x ] _Does it solve one of the open issues on github?_

### _Notes on testing_
 - _Until the automated tests pass, please mark the PR as a draft._
 - _On the XENONnT fork we test with database access, on private forks there is no database access for security considerations._

All _italic_ comments can be removed from this template.

### CODE REVIEW

File: straxen/corrections_services.py [hunk 1]
```diff
@@ -130,14 +129,22 @@ def _get_correction(self, run_id, correction, version):
...
+ 
                         if version in 'ONLINE':
                             df = self.interface.interpolate(df, when, how='fill')
                         else:
                             df = self.interface.interpolate(df, when)
                         values.append(df.loc[df.index == when, version].values[0])
             else:
-                df = self.interface.read(correction)
+                df = self.interface.read_by_index(correction, when)
+                if self.check_for_nans(when, df, version):
```
Comment (jmosbacher) [inline straxen/corrections_services.py:144]: if you use the read_by_index function the returned dataframe has only the two relevant items to this time so a simple df[version].isnull().values.any() makes more sense (if either of the two enclosing points are nan, raise error)

Comment (ahiguera-mx) [reply straxen/corrections_services.py]: yes, that make senses. thanks

**Review by ershockley — APPROVED** [review]
Thanks @ahiguera-mx! I think this is good to merge but we need to make sure it is at same time or after (strax529 )[https://github.com/AxFoundation/strax/pull/529]

**Comment by WenzDaniel** [comment]
Same comment as in https://github.com/AxFoundation/strax/pull/529#pullrequestreview-746453238

**Comment by ahiguera-mx** [comment]
As I said in previously in my opinion we need this for reprocessing, specially because the discussion regarding "to_pe" values modeling

**Comment by ershockley** [comment]
Merging this now that AxFoundation/strax#529 is merged.

**Pull request merged**